src/AdScreenerTool.js

```diff
@@ -0,0 +1,345 @@
+  const [ageRange, setAgeRange] = useState({ selectedOption: getSelectorValue(params.ageRanges,ageRangeParam)});
+  const [riskScore, setRiskScore] = useState({ selectedOption: getSelectorValue(params.riskScores,riskScoreParam)});
+  const [orderBy, setOrderBy] = useState({ selectedOption: getSelectorValue(params.orderByOptions,orderByParam)});
+  const [orderDirection, setOrderDirection] = useState({ selectedOption: getSelectorValue(params.orderDirections,orderDirectionParam)});
+  
+  const numResultsToRequest = 20;
+  const resultsOffset = useRef(0);
+  const resetOffset = () => { resultsOffset.current = 0 };
+  const incermentOffset = (i) => { resultsOffset.current = resultsOffset.current + i };
+  const decermentOffset = (i) => { if (resultsOffset.current >= i) {resultsOffset.current = resultsOffset.current - i }};
+  const [showModal, setShowModal] = useState(false);
+  const handleClose = () => setShowModal(false);
+  const handleShow = () => setShowModal(true);
+  const [showNeedLoginModal, setShowNeedLoginModal] = useState(false);
+  const handleShowNeedLoginModal = () => setShowNeedLoginModal(true);
+  const handleCloseNeedLoginModal = () => setShowNeedLoginModal(false);
+  const [isGetAdsRequestPending, setIsGetAdsRequestPending] = useState(false);
+  const [isAdDataEmpty, setIsAdDataEmpty] = useState(false);
+  const [ads, setAds] = useState([
+    // {  // This is  dummy ad, helpful for testing without loading images
+    //   funding_entity: "Funding Entity",
+    //   spend: "0-99 USD",
+    //   url: "",
+    //   impressions: "1000-5000",
+    //   archive_id: "Invalid archive_id",
+    //   demo_impression_results: [],
+    //   region_impression_results: [],
+    //   alternative_archive_ids: []
+    // }
+  ]);
+
+  const getAds = () => {
+    setIsGetAdsRequestPending(true);
+    setIsAdDataEmpty(true);
+    axios
+      .get(getAdsURL, {
+        params: {
+          startDate: startDate,
+          endDate: endDate,
+          topic: topic.selectedOption.value,
+          // Using label for region is intentional. The db stores full strings, not 2 char codes
+          region: region.selectedOption.label,
+          gender: gender.selectedOption.value,
+          ageRange: ageRange.selectedOption.value,
+          riskScore: riskScore.selectedOption.value,
+          orderBy: orderBy.selectedOption.value,
+          orderDirection: orderDirection.selectedOption.value,
+          numResults: numResultsToRequest,
+          offset: resultsOffset.current
+        },
+      })
+      .then((response) => {
+        console.log(response.data);
+        setAds(response.data);
+        setIsAdDataEmpty(response.data.length === 0);
+        setIsGetAdsRequestPending(false);
+      })
+      .catch((error) => {
+        console.log(error);
+        if (error.response && error.response.status === 401) {
+          handleShowNeedLoginModal();
+        }
```

Comment:
please add `setIsAdDataEmpty(true);` and `setIsGetAdsRequestPending(false);` here so that there is some indication of a failed request